distributed/nanny.py

```diff
@@ -851,6 +851,7 @@ async def kill(
         assert self.status in (
             Status.running,
             Status.failed,  # process failed to start, but hasn't been joined yet
+            Status.closing_gracefully,
```

Comment:
This covers a race condition where `Worker.close(nanny=True)` ran successfully up until and including the message to the nanny, and then hanged - for example because of a bad plugin or extension teardown.